File: c_src/platform/linux/elements/label.c
#include "../elements.h"
#include "common/errors.h"
#include <gtk/gtk.h>

NGHandle ng_linux_create_label(const char* text) {
    if (!text) return NULL;
    
    GtkWidget* label = gtk_label_new(text);
    gtk_widget_show(label);
    
    return (NGHandle)label;
}

void ng_linux_label_invalidate(NGHandle label) {
    if (!label) return;
    gtk_widget_queue_draw((GtkWidget*)label);
}

File: c_src/platform/ios/elements/label.m
#import "../elements.h"
#import "../utils.h"
#import "../../../common/errors.h"
#import <UIKit/UIKit.h>

NGHandle ng_ios_create_label_impl(const char* text) {
    if (!text) return NULL;
    
    UILabel* label = [[UILabel alloc] init];
    label.text = ng_ios_to_nsstring(text);
    label.textAlignment = NSTextAlignmentCenter;
    [label sizeToFit];
    
    return (__bridge_retained void*)label;
}



